libc/string_cpy.c

```diff
@@ -0,0 +1,1382 @@
+
+	/* Trying copy zero bytes */
+	TEST_ASSERT_EQUAL_PTR(NULL, memccpy(testArray, input, 'x', 0));
+	TEST_ASSERT_EQUAL_STRING(testArray, TEST_STR1);
+
+#pragma GCC diagnostic pop
+#endif
+}
+
+
+TEST(string_memccpy, big)
+{
+#ifdef __phoenix__
+	TEST_IGNORE();
+#endif
+#ifndef __phoenix__
+
+	char buff[BIG_NUMB] = { 0 };
+
+	/* Checking capability of handling big blocks of data */
+	TEST_ASSERT_EQUAL_INT(0, big_string());
+	TEST_ASSERT_NOT_NULL(memccpy(buff, long_str, '\xFF', BIG_NUMB));
+	TEST_ASSERT_EQUAL_STRING(buff, long_str);
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(buff, long_str, 256);
+
+	/*
+	 * Offset big chunk of data remembering '\xFF' will point to
+	 * char on number 256 and we want to copy only elements from point 128
+	 */
+	TEST_ASSERT_NOT_NULL(memccpy(&buff[128], long_str, '\xFF', BIG_NUMB - 128));
+	TEST_ASSERT_NOT_EQUAL(0, strcmp(buff, long_str));
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(&buff[128], long_str, 128);
+
+#endif
+}
+
+
+/*
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////
+ */
+
+
+TEST_SETUP(string_strncpy)
+{
+}
+
+
+TEST_TEAR_DOWN(string_strncpy)
+{
+}
+
+
+TEST(string_strncpy, basic)
+{
+	int num = 12345678;
+	char payload[MAX_STR_LEN] = { 0 },
+		 buff[MAX_STR_LEN] = { 0 };
+
+	TEST_ASSERT_EQUAL_STRING(TEST_STR1, strncpy(buff, TEST_STR1, sizeof(buff)));
+	TEST_ASSERT_EQUAL_STRING(TEST_STR1, buff);
+
+	/* Looking for leftovers from previous cpy */
+	TEST_ASSERT_EQUAL_STRING(TEST_STR2, strncpy(buff, TEST_STR2, sizeof(buff)));
+	TEST_ASSERT_EQUAL_STRING(TEST_STR2, buff);
+
+	memset(buff, 0, sizeof(buff));
+
+	/* Pragma truncation disable for the length of string tests */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstringop-truncation"
+
+	TEST_ASSERT_EQUAL_STRING(buff, strncpy(buff, TEST_STR1, sizeof(TEST_STR1) - 10));
+	TEST_ASSERT_EQUAL_STRING("Lorem ip", buff);
+
+	memset(buff, 0, sizeof(buff));
+	sprintf(payload, "%d", num);
+	TEST_ASSERT_EQUAL_STRING(buff, strncpy(buff, payload, 5));
+	TEST_ASSERT_EQUAL_STRING("12345", buff);
+
+	memset(buff, 0, sizeof(buff));
+	TEST_ASSERT_EQUAL_STRING(buff, strncpy(buff, TEST_STR2, 20));
+	TEST_ASSERT_EQUAL_STRING("Maecenas id commodo", buff);
+
+	/* Combine 2 strings using leftowers in buff */
+	TEST_ASSERT_EQUAL_STRING(buff, strncpy(buff, TEST_STR1, 10));
+	TEST_ASSERT_EQUAL_STRING("Lorem ipsud commodo", buff);
+
+#pragma GCC diagnostic pop
+}
+
+
+TEST(string_strncpy, ascii)
+{
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstringop-truncation"
+
+	char buff[MAX_ASCII_SET] = { 0 },
+		 ascii[MAX_ASCII_SET] = { 0 };
+	int i;
+
+	for (i = 1; i < MAX_ASCII_SET; i++) {
+		ascii[i - 1] = i;
+
+		/* This case test not only ascii set copied but different sizes of copy */
+		TEST_ASSERT_EQUAL_CHAR_ARRAY(ascii, strncpy(buff, ascii, i), i);
+	}
+
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(buff, ascii, sizeof(buff));
+
+#pragma GCC diagnostic pop
+}
+
+
+TEST(string_strncpy, append_null)
```

Comment:
It does not check the ability of appending NUL characters, but the ability of finding it. Please create a separate test case for testing append utility.